### PR TITLE
Add RabbitMQ Listener + Queue Management

### PR DESCRIPTION
**Pull request by jreynard-code**

### CODE REVIEW

File: api/src/main/resources/application.yml [hunk 1]
```diff
@@ -94,13 +99,24 @@ csm:
...
+        username: "storage_writer_username"
+        password: "storage_writer_password"
       admin:
         username: "storage_admin_username"
         password: "storage_admin_password"
     eventbus:
       host: "localhost"
-      username: "eventbus_admin_username"
-      password: "eventbus_admin_password"
+      port: "5672"
```
Comment (sjoubert) [inline api/src/main/resources/application.yml:110]: Should `port` be an integer instead of a string?

File: api/src/main/resources/application.yml [hunk 2]
```diff
@@ -94,13 +99,24 @@ csm:
...
+      default-exchange: "csm-exchange"
+      default-queue: "csm"
+      default-routing-key: "csm"
```
Comment (sjoubert) [inline api/src/main/resources/application.yml:113]: Same remark as in the common repo, this is not something currently configured on the deployment side.

File: api/src/main/resources/application.yml [hunk 3]
```diff
@@ -94,13 +99,24 @@ csm:
       reader:
         username: "storage_reader_username"
         password: "storage_reader_password"
+      writer:
```
Comment (neomatamune) [inline api/src/main/resources/application.yml:102]: The addition of the writer account could be made in a separate PR instead of having it seems like a dependency to rabbitmq

File: workspace/src/main/kotlin/com/cosmotech/workspace/service/WorkspaceServiceImpl.kt [hunk 4]
```diff
@@ -335,7 +343,10 @@ internal class WorkspaceServiceImpl(
       val pageable: Pageable =
           Pageable.ofSize(csmPlatformProperties.twincache.workspace.defaultPageSize)
       val workspaces = workspaceRepository.findByOrganizationId(organizationId, pageable).toList()
-      workspaceRepository.deleteAll(workspaces)
+      workspaces.forEach {
```
Comment (neomatamune) [inline workspace/src/main/kotlin/com/cosmotech/workspace/service/WorkspaceServiceImpl.kt:346]: Why not make use of `WorkspaceService.deleteWorkspace` that cleanly delete the workspace and its data ?

File: run/src/main/kotlin/com/cosmotech/run/service/amqp/AmqpClientServiceImpl.kt
Comment (neomatamune) [inline run/src/main/kotlin/com/cosmotech/run/service/amqp/AmqpClientServiceImpl.kt:1]: Can't seems to find what happen when an `awakeListener` happens while the associated Workspace did not go throught a `WorkspaceCreated` event (during a migration, or after a restart) how are the `Queues` created there ?

**Review by neomatamune — COMMENTED** [review]
Still have a few questions over the code, can't approve until those are answered

File: api/src/main/resources/application.yml (hunk 2 above)
Comment (jreynard-code) [reply api/src/main/resources/application.yml:104]: This is part of current Cosmotech RabbitMqConfig, will see next week to:

- remove defaultQueue creation
- change defaultExchange to amq.default (which is here as default when rabbitMQ is deployed)
- remove defaultRoutingKey

File: workspace/src/main/kotlin/com/cosmotech/workspace/service/WorkspaceServiceImpl.kt (hunk 4 above)
Comment (jreynard-code) [reply workspace/src/main/kotlin/com/cosmotech/workspace/service/WorkspaceServiceImpl.kt:350]: Because `WorkspaceService.deleteWorkspace` checks RBAC rights.
As we respond to a OrganizationUnredistered event, RBAC should be bypassed.

File: run/src/main/kotlin/com/cosmotech/run/service/amqp/AmqpClientServiceImpl.kt
Comment (jreynard-code) [reply run/src/main/kotlin/com/cosmotech/run/service/amqp/AmqpClientServiceImpl.kt:1]: I removed that and we create Queue on runStart only

**Review by neomatamune — APPROVED** [review]
Seems ok to me
